cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/aws-orchestrator/AwsOrchestrator.service.tsx

```diff
@@ -0,0 +1,44 @@
+// React imports
+import { AwsService, AWSVirtualMachine } from '../../../../../swagger-api';
+import { AWS_FIELDS } from '../../aws-mc-basic/AwsManagementClusterBasic.constants';
+import { INPUT_CHANGE } from '../../../../../state-management/actions/Form.actions';
+import { AwsDefaults } from '../default-service/AwsDefaults.service';
+import { STORE_SECTION_FORM } from '../../../../../state-management/reducers/Form.reducer';
+import { AWS_ADD_RESOURCES } from '../../../../../state-management/actions/Resources.actions';
+import { AwsResourceAction, FormAction, StoreDispatch } from '../../../../../shared/types/types';
+interface AwsOrchestratorProps {
+    awsState: { [key: string]: any };
+    awsDispatch: StoreDispatch;
+}
+export class AwsOrchestrator {
+    static async initOsImages(props: AwsOrchestratorProps) {
+        clearOsImages(props);
+        const osImages = await AwsService.getAwsosImages(props.awsState[STORE_SECTION_FORM].REGION);
+        saveOsImages(props, osImages);
+        setDefaultOsImage(props, osImages);
+    }
+}
+
+function clearOsImages(props: AwsOrchestratorProps) {
+    props.awsDispatch({
+        type: AWS_ADD_RESOURCES,
+        resourceName: 'osImages',
```

Comment:
I prefer to put the resourceName in the payload. This can make our action simpler.
I want to create a pattern to contain only few fields in the action:
1. action type is required which tells what to do. 
2. action payload contains all data which is needed for the action. 
3. action errors contains all related error info.
@miclettej @swalner-vmware @ChiChiHeHeZhou what do you guys think?

Reply:
I agree we don't want to add a lot of fields, but in this case it feels like the number of the fields is the same as for a Form Action. Form Action has `type`, `field` and `payload`, whereas resource action has `type`, `resourceName` and `payload`... feels OK to me